### PR TITLE
[Transform] refactor transform functions that utilize composite aggs

### PR DESCRIPTION
**Pull request by benwtrent**
while iterating on a new transform function that uses composite aggs, I noticed that composite agg functions can be described as:

- create composite agg
- parse its results

Many of the queries and APIs with the composite agg functions are very similar (if not exactly the same).

This commit refactors both the `pivot` and `latest` transform functions so that they are more uniform in design and common functions are extracted out.

### CODE REVIEW

**Review by przemekwitek — APPROVED** [review]
LGTM

And thanks for adding missing comments!

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java [hunk 1]
```diff
@@ -28,50 +21,46 @@
...
-    private final CompositeAggregationBuilder cachedCompositeAggregation;
-
+    /**
+     * Create a new Pivot function
+     * @param config A {@link PivotConfig} describing the function parameters
+     * @param transformId The referenced transform
+     * @param settings Any miscellaneous settings for the function
+     * @param version The version of the transform
+     */
     public Pivot(PivotConfig config, String transformId, SettingsConfig settings, Version version) {
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java:59]: nit: `transformId` got removed, can the parameter be easily removed, too?

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/common/DocumentConversionUtils.java [hunk 2]
```diff
@@ -11,34 +11,45 @@
...
-    public static IndexRequest convertDocumentToIndexRequest(Map<String, Object> document,
+    /**
+     * Convert a raw string, object map to a valid index request
+     * @param docId The document ID
+     * @param document The document contents
+     * @param destinationIndex The index where the document is to be indexed
+     * @param destinationPipeline Optional destination pipeline
+     * @return A valid {@link IndexRequest}
+     */
+    public static IndexRequest convertDocumentToIndexRequest(String docId,
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/common/DocumentConversionUtils.java:35]: nice re-factoring of the `docId`. If I remember correctly, this removes the need for `removeInternalFields`. I think the only purpose was `_id`. We might get rid of it now.

The idea of having _internal entries_ might still be a good thing, however we can add it back, once we need it (and maybe we pass a real object instead).

So, having that said, feel free to remove `removeInternalFields` (optional).

**Review by hendrikmuhs — APPROVED** [review]
nice re-factoring, LGTM!

**Pull request merged**